### PR TITLE
libgnurl: 7.54.1 -> 7.61.1.

### PR DESCRIPTION
**Pull request by royneary**
###### Motivation for this change
Bump to the latest version of libgnurl. In the previous version of the package HTTPS support was not compiled in.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libgnurl/default.nix [hunk 1]
```diff
@@ -1,38 +1,30 @@
...
 
-  nativeBuildInputs = [ autoreconfHook ];
-  buildInputs = [ perl gnutls gss openssl zlib libidn ];
-
-  preConfigure = ''
-    sed -e 's|/usr/bin|/no-such-path|g' -i.bak configure
-  '';
+  nativeBuildInputs = [ libtool groff perl pkgconfig python2 ];
+    
+  buildInputs = [ gnutls zlib libidn ];
```
Comment (xeji) [inline pkgs/development/libraries/libgnurl/default.nix]: Looks like IDN support doesn't work with `libidn` anymore, see the configure log. You need `libidn2` instead (and this requires `libunistring` as well).

Also I suggest to add  `nghttp2` here to enable HTTP2 support.

File: pkgs/development/libraries/libgnurl/default.nix [hunk 2]
```diff
@@ -1,38 +1,30 @@
...
 stdenv.mkDerivation rec {
-  version = "7.54.1";
+  version = "7.61.1";
 
   name = "libgnurl-${version}";
 
   src = fetchurl {
-    url = "https://gnunet.org/sites/default/files/gnurl-${version}.tar.bz2";
-    sha256 = "0szbj352h95sgc9kbx9wzkgjksmg3g5k6cvlc7hz3wrbdh5gb0a4";
+    url = "https://ftp.gnu.org/gnu/gnunet/gnurl-${version}.tar.gz";
```
Comment (xeji) [inline pkgs/development/libraries/libgnurl/default.nix]: Please use `url = "mirror://gnu/gnunet/..."` here. This automatically uses a mirror list provided in nixpkgs, see [here](https://github.com/NixOS/nixpkgs/blob/d35c7f9a914c17fcca4839593250d906de1afa24/pkgs/build-support/fetchurl/mirrors.nix#L29).

File: pkgs/development/libraries/libgnurl/default.nix [hunk 3]
```diff
@@ -1,38 +1,30 @@
...
+  buildInputs = [ gnutls zlib libidn ];
 
   configureFlags = [
-    "--enable-ipv6" "--with-gnutls" "--without-libmetalink" "--without-winidn"
-    "--without-librtmp" "--without-nghttp2" "--without-nss" "--without-cyassl"
-    "--without-polarssl" "--without-ssl" "--without-winssl"
-    "--without-darwinssl" "--disable-sspi" "--disable-ntlm-wb" "--disable-ldap"
-    "--disable-rtsp" "--disable-dict" "--disable-telnet" "--disable-tftp"
-    "--disable-pop3" "--disable-imap" "--disable-smtp" "--disable-gopher"
-    "--disable-file" "--disable-ftp" "--disable-smb"
```
Comment (xeji) [inline pkgs/development/libraries/libgnurl/default.nix]: Removing these is fine, they're the default settings anyway.

**Comment by royneary** [comment]
Fixed as you suggested.

**Comment by xeji** [comment]
@GrahamcOfBorg build libgnurl

**Comment by xeji** [comment]
@GrahamcOfBorg build gnunet

**Pull request merged**